third_party/2and3/werkzeug/utils.pyi

```diff
@@ -31,7 +31,15 @@ def format_string(string, context): ...
 def secure_filename(filename): ...
 def escape(s, quote: Optional[Any] = ...): ...
 def unescape(s): ...
-def redirect(location, code: int = ..., Response: Optional[Any] = ...) -> Response: ...
+
+# 'redirect' returns a werkzeug Response, unless you give it
+# another Response type to use instead.
+ResponseClass = TypeVar("ResponseClass", bound=Response)
+@overload
+def redirect(location, code: int = ...) -> Response: ...
+@overload
+def redirect(location, code: int = ..., Response: Type[ResponseClass] = ...) -> ResponseClass: ...
```

Comment:
Since `None` is a valid value for `Response`, it's better to leave the `Response` argument in the first overload with a `None` type and remove the default from the second overload:

```python
@overload
def redirect(location, code: int = ..., Response: None = ...) -> Response: ...
@overload
def redirect(location, code: int = ..., Response: Type[ResponseClass]) -> ResponseClass: ...
```

Reply:
It doesn't compile (default arg follows non-default arg) unless I also leave the default in the 2nd overload. I did that and it seems to work though.

Thanks for the feedback!